### DISCUSSION TITLE
BMS parameters considering the PV power

### DISCUSSION THREAD

**Discussion by Micha854** [Louisvdw/dbus-serialbattery#155] [General]
hello dear louis,

you can adopt the following feature, an example:

PV power delivers 500 watts. 400 watts is consumed. CCL is limited to 5A because the battery is almost full.

there would be enough pv power to supply consumers and 100 watts to fully charge the battery.

Now the following problem arises:

Since CCL is limited to 5A, only about 100 watts comes from the PV. that is not enough to cover consumption. accordingly, the battery is always discharge....

I hope you understand the problem. one would have to consider the current PV power. can you talk about it?

lg
micha

**Comment by Louisvdw** [comment]
Hi Micha
The CCL limit is only a battery limit. It is not a system limit.
So what happens is that the battery say I can take 5A of charge and the GX will try to keep to that 5A limit. It will instruct everything that can give a charge to give less power so that the battery will only get a charge of 5A or less, but any consumption that the inverter has is excluded from that. So if the system is using 10A then your MPPT will be asked to keep it below 10+5 A charge. (consumption plus battery charge). 
You  will see the MPPT power change all the time to match this, but it cannot change instantly and has to ramp up an down to the level. The battery in this case is used to even this out as it can change faster. This is how the victron GX ESS system has been designed and the driver has not control over this. 
The only suggestion I have is to perhaps increase your grid set point.

**Comment by Micha854** [comment]
Hello, I have to say that I do not have an ESS system. Only a phoenix 24/1200 inverter is running here and the scenario you describe does not occur. I then only have 5A available no matter what else is consumed

**Reply by Louisvdw** [reply]
Intresting. I wonder if the GX handles this differently when a Phoenix or Multiplus is connected.
If you want to disable CCCM and see if that helps there is a beta option available. Edit your /data/etc/dbus-serialbattery/utils.py and set CCCM_ENABLE = False (around line 25) and restart

**Comment by Micha854** [comment]
hey louis,

yes it will look really different without ESS system. here is a screenshot of the PV charger
<img src="https://i.postimg.cc/8zm2Bnkm/Screenshot-2022-07-26-085339.png">

if you need more screens please let me know. I am now testing CCCM_ENABLE = False

**Reply by Louisvdw** [reply]
In an ESS system that is also what you will see. It migh make a bit more sense when you look at the PV Yield graph instead. The power output of the PV system is reduced to match the demand. You will see the same if your system is full and has reached the max battery voltage. Even with CCCM disabled and even if you don't have the battery driver or connection. That is how the Victron system is designed to work.

Here is my yield graph with the matching voltage current graph. Around 1pm the batteries are full and the last large loads has stopped, so the system is reducing the yield of the PV to only produce what is consumed. The nice round climing curve starts to get spikey. That is what you want to see in your system. 
![image](https://user-images.githubusercontent.com/26077665/181776148-59862c01-b70b-48f6-a1b3-1dc0098d1d54.png)

**Comment by Micha854** [comment]
the option CCCM_ENABLE = False ensures that the CCL is not changed. the value remains at 50A

I think reducing the charging current is a good thing, but then with consideration of the PV power as you know it from an ESS system.

Can you tell me what I would have to adjust for this? There must be a problem with your code?
